src/bulls_and_cows_lib/game.cpp

```diff
@@ -14,47 +15,177 @@ namespace bulls_and_cows {
-               "    Display the board and the list of attempts so far\n"
-               "    Display a message telling if the computer won or lost\n";
+        /* std::cout
+             << "TODO:\n"
+                "    Create a board with a randomly generated secret code\n"
+                "    Generate the list of all the possible codes\n"
+                "    DO\n"
+                "       Display the board and the list of attempts so far\n"
+                "       Display the number of remaining possible codes so far\n"
+                "       Wait for 2 seconds\n"
+                "       Pick a random attempt among in the list of remaining possible codes\n"
+                "       Compare the computer's attempt with the secret code and deduce the number of bulls and cows\n"
+                "       Add the computer's attempt to the list of attempts of the board\n"
+                "       Remove all the codes that are incompatible with the attempt's feedback from the list of "
+                "possible codes\n"
+                "    WHILE not end of game\n"
+                "    Display the board and the list of attempts so far\n"
+                "    Display a message telling if the computer won or lost\n";*/
+
+        Board board = create_board(game_options);
+        Board compboard;
+
+        do
+        {
+            display_board(std::cout, game_options, board);
+
+            compboard = create_board(game_options);
+
+            AttemptAndFeedback att;
+            att.attempt = compboard.secret_code;
+            att.feedback = compare_attempt_with_secret_code(att.attempt, board.secret_code);
+
+            board.attempts_and_feedbacks.push_back(att);
+
+        } while (!is_end_of_game(game_options, board));
+
+        is_end_of_game(game_options, board);
+        is_win(game_options, board);
     }
 
     void configure_game_options(GameOptions& game_options)
     {
-        std::cout << "TODO:\n"
-                     "    DO\n"
-                     "       Display the current game options\n"
-                     "       Display the game options menu\n"
-                     "       Ask the user to type its choice\n"
-                     "       Treat the user's choice\n"
-                     "    UNTIL user's choice is to go back to main menu\n";
+        /* std::cout << "TODO:\n"
+                      "    DO\n"
+                      "       Display the current game options\n"
+                      "       Display the game options menu\n"
+                      "       Ask the user to type its choice\n"
+                      "       Treat the user's choice\n"
+                      "    UNTIL user's choice is to go back to main menu\n";*/
+
+        bool verif = false;
+
+        do
+        {
+            display_game_options(std::cout, game_options);
+            display_game_options_menu(std::cout);
+
+            const auto Choice = ask_game_options_menu_choice(std::cin);
+
```

Comment:
Il faudrait que out soit déclaré dans le corps du case GameOptionsMenuChoice::SaveOptions.
Mais il faudrait aussi que le corps des case GameOptionsMenuChoice::SaveOptions et case case GameOptionsMenuChoice::LoadOptions contiennent des accolades pour que la durée de vie de out et in soit celles des accolades.

Reply:
Done